### ISSUE TITLE
operation for dictionary attempted on object of type null: returning null for attempted key retrieval

### ISSUE THREAD

**Issue by nemobis** [ocrmypdf/OCRmyPDF#934]
**Describe the bug**

ocrmypdf terminates with an exception: pikepdf._qpdf.PdfError: operation for dictionary attempted on object of type null: returning null for attempted key retrieval

**To Reproduce**

```bash
ocrmypdf --skip-text in.pdf out.pdf
```

Typical output:
```
An exception occurred while executing the pipeline
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/ocrmypdf/_sync.py", line 385, in run_pipeline
    exec_concurrent(context, executor)
  File "/usr/local/lib/python3.10/site-packages/ocrmypdf/_sync.py", line 297, in exec_concurrent
    pdf = ocrgraft.finalize()
  File "/usr/local/lib/python3.10/site-packages/ocrmypdf/_graft.py", line 198, in finalize
    self.pdf_base.save(self.output_file)
  File "/usr/local/lib64/python3.10/site-packages/pikepdf/_methods.py", line 774, in save
    self._save(
pikepdf._qpdf.PdfError: operation for dictionary attempted on object of type null: returning null for attempted key retrieval
```
[ocr.log](https://github.com/ocrmypdf/OCRmyPDF/files/8336895/ocr.log)

**Example file**

I have a few, I need to pick suitable examples.

This seems to be https://github.com/qpdf/qpdf/issues/527 ; I'm still on qpdf 10.3. However, I can't reproduce with the two test cases over there.

**Expected behavior**

Ignore the error.

**System**

```
$ python --version
Python 3.10.2
$ ocrmypdf --version
13.4.1
$ rpm -qa | grep -E "^(qpdf|ghostscript|ocrmypdf|jbig2|python3-pike|tesseract)"
tesseract-tessdata-doc-4.1.0-2.fc35.noarch
tesseract-langpack-eng-4.1.0-2.fc35.noarch
tesseract-langpack-fin-4.1.0-2.fc35.noarch
tesseract-4.1.3-1.fc35.x86_64
tesseract-langpack-ita-4.1.0-2.fc35.noarch
jbig2dec-libs-0.19-5.fc35.x86_64
qpdf-libs-10.3.2-2.fc35.x86_64
qpdf-10.3.2-2.fc35.x86_64
jbig2dec-devel-0.19-5.fc35.x86_64
jbig2dec-0.19-5.fc35.x86_64
ghostscript-tools-fonts-9.55.0-2.fc35.x86_64
ghostscript-tools-printing-9.55.0-2.fc35.x86_64
ghostscript-9.55.0-2.fc35.x86_64
tesseract-langpack-chi_tra_vert-4.1.0-2.fc35.noarch
tesseract-langpack-chi_tra-4.1.0-2.fc35.noarch
tesseract-langpack-chi_sim_vert-4.1.0-2.fc35.noarch
tesseract-langpack-chi_sim-4.1.0-2.fc35.noarch
tesseract-langpack-deu-4.1.0-2.fc35.noarch
```

**Comment by nemobis** [comment]
A workaround that works for me at the moment is to run `mutool convert -O sanitize -o out.pdf in.pdf` before runnyng ocrmypdf.

https://mupdf.com/docs/manual-mutool-convert.html

**Comment by nemobis** [comment]
An example is downloadable at https://doi.org/10.1515/9783111393551-fm (and attached: [10.1515_9783111393551-toc.pdf](https://github.com/ocrmypdf/OCRmyPDF/files/8337053/10.1515_9783111393551-toc.pdf) ).

```
$ ocrmypdf --skip-text /tmp/10.1515_9783111393551-toc.pdf /tmp/out.pdf
Scanning contents: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 18.32page/s]
Start processing 2 pages concurrently
    1 skipping all processing on this page                                                                                                                                                                                                  
    2 skipping all processing on this page                                                                                                                                                                                                  
OCR: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2.0/2.0 [00:00<00:00, 819.60page/s]
An exception occurred while executing the pipeline
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/ocrmypdf/_sync.py", line 385, in run_pipeline
    exec_concurrent(context, executor)
  File "/usr/local/lib/python3.10/site-packages/ocrmypdf/_sync.py", line 297, in exec_concurrent
    pdf = ocrgraft.finalize()
  File "/usr/local/lib/python3.10/site-packages/ocrmypdf/_graft.py", line 198, in finalize
    self.pdf_base.save(self.output_file)
  File "/usr/local/lib64/python3.10/site-packages/pikepdf/_methods.py", line 774, in save
    self._save(
pikepdf._qpdf.PdfError: operation for dictionary attempted on object of type null: returning null for attempted key retrieval
```

**Comment by jbarlow83** [comment]
This should be fixed by upgrading to pikepdf 5.1.3. See pikepdf issue 317.
